libsolidity/analysis/PostTypeChecker.cpp

```diff
@@ -422,6 +433,60 @@ struct ReservedErrorSelector: public PostTypeChecker::Checker
 	}
 };
 
+class LValueChecker: public ASTConstVisitor
+{
+public:
+	LValueChecker(Identifier const& _identifier): m_declaration(_identifier.annotation().referencedDeclaration) {}
+	bool willBeWrittenTo() const { return m_willBeWrittenTo; }
+	void endVisit(Identifier const& _identifier) override
+	{
+		if (
+			_identifier.annotation().referencedDeclaration == m_declaration &&
```

Comment:
I don't think this code will work as expected if `referencedDeclaration` happens to be null. It's probably never supposed to be null, but in that case you should assert that here.

Then we could also safely use `ASTNode`'s `==` instead of comparing pointers:
```suggestion
			*_identifier.annotation().referencedDeclaration == *m_declaration &&
```

Reply:
Done.